torch/_inductor/codegen/cpp.py

```diff
@@ -2863,6 +2866,11 @@ def finalize_kernel(self, new_kernel, nodes):
         ws = self.ws
         new_kernel.codegen_loops(code, ws)
 
+    def get_num_args(self):
```

Comment:
Rather then adding this function to the scheduler, can we add it specifically to the cpp backend. I think codegen_nodes is a natural place for this, it's how we handle it for foreach codegen. I don't see why the scheduler needs to be involved at all.

Reply:
@mlazos is the arg number limit an issue impacting triton codegen in general or only on specific cases like "foreach"? What we tried to do here is to follow your idea of splitting kernels in "foreach" codegen but in a more general way, considering one could construct an arbitrary case with a lot of arguments.

Reply:
You can only split foreach kernels like this because in general you can't unsplit an arbitrary kernel. With foreach you're okay because you're splitting across subkernels.

Reply:
> You can only split foreach kernels like this because in general you can't split an arbitrary kernel. With foreach you're okay because you're splitting across subkernel boundaries. Since the fules differ for cpp and triton I feel like it makes more sense to have each codegen backend know how to partition a codegen schedule

@mlazos We are not trying to split a generated kernel but control the number of args from the generated kernel. Foreach kernel is a special case since you would have a lot of sub-nodes and a lot of args inside. On the other hands, if the optimizer is not models as foreach, would it become a problem for Triton codegen too? The approach in this PR is orthogonal to the one in Foreach kernel and handles things not covered by Foreach nodes.

Reply:
@jansel any comments for this discussion?

Reply:
We can discuss in the meeting, but we don't hit these limits in a general kernel because we limit the number of fusions to 64 in the general case.